pkg/utils/kubernetes/kubectl/kustomize.go

```diff
@@ -0,0 +1,58 @@
+package kubectl
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"sigs.k8s.io/kustomize/api/krusty"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/filesys"
+)
+
+// GetKustomizedManifest takes a kustomization and any number of manifest readers. It adds the manifests to the
+// kustomization's resources and returns a reader with the rendered kustomization.
+func GetKustomizedManifest(kustomization types.Kustomization, manifests ...io.Reader) (io.Reader, error) {
+	workDir, err := os.MkdirTemp("", "ktf.")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(workDir)
+	for i, manifest := range manifests {
+		orig, err := io.ReadAll(manifest)
+		if err != nil {
+			return nil, err
+		}
+		err = os.WriteFile(filepath.Join(workDir, fmt.Sprintf("base_%d.yaml", i)), orig, 0o600) //nolint:gomnd
+		if err != nil {
+			return nil, err
+		}
+		kustomization.Resources = append(kustomization.Bases, fmt.Sprintf("base_%d.yaml", i))
```

Comment:
1. Why assign to `kustomization.Resources` but append to `kustomization.Bases`?
2. `bases` was deprecated in v2.1.0 https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/bases/

Reply:
It was originally written using the latter (a lot of the Kustomize docs examples still use bases) and my brain is conflating the two :shrug: